apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -2203,26 +2235,27 @@ private static TypeName selectJavaFieldType(
       boolean isColumnOptional,
       Optional<Integer> columnLength,
       Optional<Integer> columnScale) {
-    if (type == RifColumnType.CHAR
-        && columnLength.orElse(Integer.MAX_VALUE) == 1
-        && !isColumnOptional) return TypeName.CHAR;
-    else if (type == RifColumnType.CHAR
-        && columnLength.orElse(Integer.MAX_VALUE) == 1
-        && isColumnOptional) return ClassName.get(Character.class);
-    else if (type == RifColumnType.CHAR) return ClassName.get(String.class);
-    else if (type == RifColumnType.DATE && columnLength.orElse(0) == 8)
-      return ClassName.get(LocalDate.class);
-    else if (type == RifColumnType.TIMESTAMP && columnLength.orElse(0) == 20)
-      return ClassName.get(Instant.class);
+    if (type == RifColumnType.CHAR) {
+      if (columnLength.orElse(Integer.MAX_VALUE) == 1) {
+        return isColumnOptional ? ClassName.get(Character.class) : TypeName.CHAR;
+      } else {
+        return ClassName.get(String.class);
+      }
+    } else if (type == RifColumnType.DATE) return ClassName.get(LocalDate.class);
+    else if (type == RifColumnType.TIMESTAMP) return ClassName.get(Instant.class);
     else if (type == RifColumnType.NUM && columnScale.orElse(Integer.MAX_VALUE) > 0)
       return ClassName.get(BigDecimal.class);
     else if (type == RifColumnType.NUM
         && columnScale.orElse(Integer.MAX_VALUE) == 0
         && !isColumnOptional) return TypeName.INT;
-    else if (type == RifColumnType.NUM
-        && columnScale.orElse(Integer.MAX_VALUE) == 0
-        && isColumnOptional) return ClassName.get(Integer.class);
-    else throw new IllegalArgumentException("Unhandled field type: " + type.name());
+    else if (type == RifColumnType.SMALLINT) {
+      return isColumnOptional ? ClassName.get(Short.class) : TypeName.SHORT;
```

Comment:
What is this even doing? This method could particularly use an @returns in the doc cause I don't know whats even happening here. So we return "Short" or "short" based on.. the column being optional? Not a condemnation of this change since its following the convention of the method, but its just written confusingly in the first place.

Reply:
So the TypeName that it returns is a Java Poet ORM data type to be applied to that column in the entity model. I've added a return to try to explain a bit more. And yes your are right on many counts....it is confusing (to say the least) and the isColumnOptional drives whether the column can be a primitive or a Java class.